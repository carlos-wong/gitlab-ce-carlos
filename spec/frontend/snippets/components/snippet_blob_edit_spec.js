import SnippetBlobEdit from '~/snippets/components/snippet_blob_edit.vue';
import BlobHeaderEdit from '~/blob/components/blob_edit_header.vue';
import BlobContentEdit from '~/blob/components/blob_edit_content.vue';
import { shallowMount } from '@vue/test-utils';

jest.mock('~/blob/utils', () => jest.fn());

describe('Snippet Blob Edit component', () => {
  let wrapper;
  const content = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
  const fileName = 'lorem.txt';

  function createComponent() {
    wrapper = shallowMount(SnippetBlobEdit, {
      propsData: {
        content,
        fileName,
      },
    });
  }

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  describe('rendering', () => {
    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders required components', () => {
      expect(wrapper.contains(BlobHeaderEdit)).toBe(true);
      expect(wrapper.contains(BlobContentEdit)).toBe(true);
    });
  });
});
