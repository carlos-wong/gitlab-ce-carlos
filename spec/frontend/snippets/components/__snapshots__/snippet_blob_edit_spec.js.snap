// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet Blob Edit component rendering matches the snapshot 1`] = `
<div
  class="form-group file-editor"
>
  <label>
    File
  </label>
   
  <div
    class="file-holder snippet"
  >
    <blob-header-edit-stub
      value="lorem.txt"
    />
     
    <blob-content-edit-stub
      filename="lorem.txt"
      value="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
    />
  </div>
</div>
`;
